decoder/expr_literal_type_hover.go

```diff
@@ -0,0 +1,149 @@
+package decoder
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func (lt LiteralType) HoverAtPos(ctx context.Context, pos hcl.Pos) *lang.HoverData {
+	typ := lt.cons.Type
+
+	if typ == cty.DynamicPseudoType {
+		val, _ := lt.expr.Value(nil)
+		if val.IsKnown() && !val.IsNull() {
+			typ = val.Type()
+		}
```

Comment:
```suggestion
		val, diags := lt.expr.Value(nil)
		if !diags.HasErrors() {
			typ = val.Type()
		}
```
Should be more or less equivalent, but this is probably more correct.